portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -37,12 +37,22 @@
 
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        System.out.println("hey, whats up !");
+        System.out.println(request);
+
         Query query = new Query("Comments").addSort("timestamp", SortDirection.DESCENDING);
 
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
         PreparedQuery results = datastore.prepare(query);
+        int result_length =0;
+
+        for (Entity commentEntity : results.asIterable()) {
+            result_length++;
+        }
 
         List<Comments> comments = new ArrayList<>();
+        int no_of_comments = Integer.parseInt(request.getParameter("no"));
```

Comment:
This isn't a very intuitive variable name. 'no' can mean 'No' or 'Number'.

I would call it comment_count_target